src/ray/common/client_connection.cc

```diff
@@ -410,32 +410,32 @@ void ServerConnection::DoAsyncWrites() {
 }
 
 std::shared_ptr<ClientConnection> ClientConnection::Create(
-    MessageHandler &message_handler,
+    const MessageHandler &message_handler,
+    const ConnectionErrorHandler &connection_error_handler,
```

Comment:
Take value instead of constant reference.
Reference: effective modern C++ item-41.